lightningd/gossip_control.c

```diff
@@ -212,9 +234,16 @@ static void handle_private_update_data(struct lightningd *ld, const u8 *msg)
 		fatal("Gossip gave bad GOSSIPD_REMOTE_CHANNEL_UPDATE %s",
 		      tal_hex(msg, msg));
 	channel = any_channel_by_scid(ld, &update->scid, true);
+	/* 0conf channels can pass the peer's channel alias */
+	if (!channel) {
+		channel = find_channel_by_remote_alias(ld, &update->scid,
+						       &update->source_node);
+	}
```

Comment:
This is wrong, and means there's a bug somewhere else.

We can't use the remote alias, it's not unique (Bob and Carol can give Alice the same alias).  But Bob and Carol should always send messages to Alice using *ALICE'S* alias.  It's quite possible we get this wrong somewhere...